src/topp/IDFilter.cpp

```diff
@@ -479,6 +506,39 @@ class TOPPIDFilter :
       filter.filterPeptideEvidences(peptides);
     }
 
+    Int min_cleavages = getIntOption_("missed_cleavages:min");
+    Int max_cleavages = getIntOption_("missed_cleavages:max");
+    
+    if (not (min_cleavages == -1 && max_cleavages == -1))
+    {
+      LOG_INFO << "Filtering peptide hits by their missed cleavages count..." << endl;
+
+      // Configure Enzymatic digestion
+      EnzymaticDigestion digestion;
+      String enzyme = getStringOption_("missed_cleavages:enzyme").trim();
+      if (!enzyme.empty())
+      {
+        digestion.setEnzyme(enzyme);
+      }
+
+      Int min_cleavages = getIntOption_("missed_cleavages:min");
+      Int max_cleavages = getIntOption_("missed_cleavages:max");
+      
+      // Build the digest filter function
+      IDFilter::PeptideDigestionFilter filter(digestion);
+      filter.setMaxCleavagesAllowed(max_cleavages);
+      filter.setMinCleavagesAllowed(min_cleavages);
+
+      // Filter peptide hits      
+      for (auto& peptide : peptides)
+      {
+        auto hits = peptide.getHits();
+        filter.filterPeptideSequences(hits);
```

Comment:
`filter.filterPeptideSequences(peptide.getHits())` should do (since it will modify in-place)?

Reply:
It violates its [const property](https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/METADATA/PeptideIdentification.h#L126).